### PR TITLE
add --fn-user flag to function run

### PR DESCRIPTION
**Pull request by Shell32-Natsu**
https://github.com/GoogleContainerTools/kpt/issues/939

### CODE REVIEW

File: cmd/config/internal/commands/run_test.go [hunk 1]
```diff
@@ -283,6 +285,26 @@ apiVersion: v1
 			args: []string{"run", "dir", "--image", "foo:bar", "--", "a=b", "c", "e=f"},
 			err:  "must have keys and values separated by",
 		},
+		{
+			name: "specify username",
+			args: []string{"run", "dir", "--fn-user", "root", "--image", "foo:bar"},
```
Comment (prachirp) [inline cmd/config/internal/commands/run_test.go:290]: Out of curiosity, what happens if --fn-user is passed twice with different values?

**Comment by Shell32-Natsu** [comment]
> This adds fn-user for imperative runs. We also need it for declarative runs.

Declarative way has been implemented in https://github.com/kubernetes-sigs/kustomize/pull/2871

File: cmd/config/internal/commands/run_test.go (hunk 1 above)
Comment (Shell32-Natsu) [reply cmd/config/internal/commands/run_test.go:290]: That's an interesting question. It's handled by cobra library. I did a quick experiment and it looks like the latter one will take effect.

**Comment by prachirp** [comment]
@Shell32-Natsu Got it thanks

/lgtm

**Pull request merged**